sklearn/neighbors/_base.py

```diff
@@ -619,6 +619,29 @@ def _fit(self, X, y=None):
                 else:
                     self._fit_method = "brute"
 
+        if (
+            # TODO(1.3): remove "wminkowski"
+            self.effective_metric_ in ("wminkowski", "minkowski")
+            and self.effective_metric_params_["p"] < 1
+        ):
+            # For 0 < p < 1 Minkowski distances aren't valid distance
+            # metric as they do not satisfy triangular inequality:
+            # they are semimetrics.
+            # algo="kd_tree" and algo="ball_tree" can't be used because
+            # KDTree and BallTree require distance metric to work properly.
+            # However, the brute-force algorithm supports semimetrics.
+            if self._fit_method == "brute":
+                warnings.warn(
+                    "Mind that for 0 < p < 1, Minkowski metrics are not distance"
+                    " metric. Continuing the execution with `algo='brute'`."
```

Comment:
```suggestion
                    " metrics. Continuing the execution with `algo='brute'`."
```

Reply:
I would remove `Continuing the execution with algo="brute"` since one would have the impression that we change `algo`.